### DISCUSSION TITLE
Brew apparently requires a nonexistent xcode version?

### DISCUSSION THREAD

**Discussion by randomairborne** [Homebrew/discussions#3822] [Everyday usage] [answered]
When I run `brew upgrade, I repeatedly get ```Error: Your Xcode (14.0.1) is too outdated.
Please update to Xcode 14.1 (or delete it).
Xcode can be updated from the App Store.

Error: Your Command Line Tools (CLT) does not support macOS 13.
It is either outdated or was modified.
Please update your Command Line Tools (CLT) or delete it if no updates are available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.1.```
This is... kind of silly, as I currently have xcode 14.0.1 which supports MacOS 13 just fine, and as of today 14.1 is still a release candidate. Is there a way to fix this?

**Comment by carlocab** [comment, marked as answer]
Please install the release candidate. Xcode 14.0.1 does not include the SDK for Ventura.

**Reply by p-linnane** [reply]
To add onto this: we expected Apple to release Xcode 14.1 today with Ventura as they normally do. Instead, they gave us a second release candidate which is unusual to say the least.

**Reply by crupest** [reply]
I encountered the same problem after updating to Ventura and spent quite a lot of time to figure out this wired fact.

It's like _Apple_ give you the new system to __use__ without giving you the tools to __develop__ for it. Couldn't even image an OS with no SDK. 🤣

**Reply by tiernano** [reply]
installed RC2 of 14.1, still getting the same error... RC2 was released today (yesterday?) and brew doctor gives the same error, same with brew upgrade...

**Reply by p-linnane** [reply]
Did you finish the install by opening the new version of Xcode or running `xcodebuild -runFirstLaunch`?

**Reply by JunJCode** [reply]
I was getting the same errors but after `xcode-select --install`, everything is working. But I run brew doctor, it's throwing me that same error message but everything is working.

However, is this normal? In the SDKs folder there is five folders:
MacOSX.sdk (Alias), MacOSX12.0.sdk, MacOSX12.1.sdk, MacOSX12.3.sdk, MacOSX12.sdk (Alias).

**Reply by smenor** [reply]
Seconding the still unable to do this in RC2 of 14.1

**Reply by net** [reply]
Here's what worked for me:

1. Download Xcode 14.1 RC2 from https://developer.apple.com/download/all/?q=Xcode.
2. Unzip Xcode_14.1_Release_Candidate_2.xip.
3. Rename the unzipped Xcode app to "Xcode_14_1". This will allow it to co-exist next to the App Store Xcode in /Applications.
4. Move Xcode_14_1 into /Applications.
5. Open Xcode_14_1 and complete the setup for Mac development (I'm not sure if this is necessary).
6. **Run `sudo xcode-select --switch /Applications/Xcode_14_1.app`**.
7. Run `brew doctor` and see no Xcode warnings.

**Reply by Bo98** [reply]
> I was getting the same errors but after `xcode-select --install`, everything is working. But I run brew doctor, it's throwing me that same error message but everything is working.
> 
> However, is this normal? In the SDKs folder there is five folders: MacOSX.sdk (Alias), MacOSX12.0.sdk, MacOSX12.1.sdk, MacOSX12.3.sdk, MacOSX12.sdk (Alias).

That appears to be 14.0 command line tools rather than 14.1 since you are missing MacOSX13.sdk, so you may still experience errors. I recommend downloading "Command Line Tools for Xcode 14.1 Release Candidate 2" from developer.apple.com still.

But yes, the command line tools installer is not very good at clearing up older SDKs. If you want to clean that up, you can delete the whole `/Library/Developer/CommandLineTools` installation and reinstall.

**Reply by lenincarlen** [reply]
> > Estaba recibiendo los mismos errores, pero después, todo está funcionando. Pero ejecuto brew doctor, me está lanzando el mismo mensaje de error, pero todo está funcionando.`xcode-select --install`
> > Sin embargo, ¿es esto normal? En la carpeta SDK hay cinco carpetas: MacOSX.sdk (Alias), MacOSX12.0.sdk, MacOSX12.1.sdk, MacOSX12.3.sdk, MacOSX12.sdk (Alias).
> 
> Eso parece ser herramientas de línea de comandos 14.0 en lugar de 14.1 ya que falta MacOSX13.sdk, por lo que aún puede experimentar errores. Recomiendo descargar "Command Line Tools for Xcode 14.1 Release Candidate 2" de developer.apple.com todavía.
> 
> Pero sí, el instalador de herramientas de línea de comandos no es muy bueno para limpiar SDK más antiguos. Si desea limpiar eso, puede eliminar toda la instalación y volver a instalarla.`/Library/Developer/CommandLineTools`

Me funciono perfectamente. Gracias.

**Reply by andrewcrook** [reply]
A I missing something? 
is SDK for Ventura actually required for brew to run on Ventura?
I haven't developed much with Xcode tools and I presumed the previous SDK would still work just not supporting new features that brew probably doesn’t need. I mean generally most apps don't require recompiled under the SDK for specific OS versions.

**Reply by Bo98** [reply]
It depends a bit on what you're doing but yeah, if you do a fresh install of `brew install gcc` and try use GCC to compile basic programs then it will be very broken without the matching SDK.

Other things can vary a bit: Python can be mostly ok except for few things that might build native code on a `pip install`, depending on how SDK selection works for individual PyPI packages.

`pkg-config` is hard tied to the SDK, but how much that affects things totally varies depending on how much Apple actually updated in a new SDK. We do try avoid forcing any particular SDK there however.

When building any formula from source: we hard require a matching SDK regardless. You'll see stricter `brew doctor` checks kick in on that scenario, which verifies the contents of SDKs.

**Reply by commjoen** [reply]
https://github.com/orgs/Homebrew/discussions/3822#discussioncomment-3965318 worked for me! Thank you @net !

**Reply by romanr** [reply]
Since the only guaranteed way to make it work is to have pre-release XCode installed, shouldn't the message "`Xcode can be updated from the App Store.`" be changed to "`Register Apple Developer account and download XCode 14.1 beta`" ?

**Reply by smenor** [reply]
> Since the only guaranteed way to make it work is to have pre-release XCode installed, shouldn't the message "`Xcode can be updated from the App Store.`" be changed to "`Register Apple Developer account and download XCode 14.1 beta`" ?
> 
> 

Totally agree with that ∧

I have a developer account and the β Xcode and still had to be told that I explicitly needed to get the prerelease command line tools and that they weren't the ones included in / accessible from the β Xcode and I'm sure I'm far from alone

**Reply by andrewcrook** [reply]
> Since the only guaranteed way to make it work is to have pre-release XCode installed, shouldn't the message "`Xcode can be updated from the App Store.`" be changed to "`Register Apple Developer account and download XCode 14.1 beta`" ?
> 
> 

You only need command line tools for Xcode which isn't on AppStore, however, brew could offer to install it.

**Reply by SMillerDev** [reply]
brew couldn't since it requires a developer account

**Reply by ZhongRuoyu** [reply]
XCode 14.1 is officially released.

**Comment by ksamc** [comment]
For those that only had the CLT installed, I did the following which worked.

1. Uninstall the existing CLT
```sudo rm -rf /Library/Developer/CommandLineTools```

2. Download the Command Line Tools for Xcode 14.1 Release Candidate 2 (latest as of writing) from here: https://developer.apple.com/download/all/?q=Command%20Line%20Tools%20for%20Xcode

3. Install the downloaded package

4. `brew upgrade`

**Reply by nwhitmont** [reply]
I'm logged in with my Apple ID, but I'm unable to load the direct link to the Command Line Tools installer.  It just redirects to my account page and says my Apple developer account is expired.  Do you need a paid developer account to download this package?

The only other option I see is to download the full XCode @7+ GB

I've never had any problems with Homebrew until this incident.  Now I need to access login protected dependency files just to run Homebrew? Seems like something went wrong in the planning process for macOS Venture 13 release.

**Reply by SMillerDev** [reply]
> Do you need a paid developer account to download this package?

You do not

> I've never had any problems with Homebrew until this incident. Now I need to access login protected dependency files just to run Homebrew? Seems like something went wrong in the planning process for macOS Venture 13 release.

Yeah, but not at Homebrew. Apple releasing an incomplete OS is something none of the Homebrew volunteers could have predicted

**Reply by Bo98** [reply]
To be clear here: MacPorts also has the exact same issue. This is not just affecting Homebrew.

**Reply by luckman212** [reply]
It was a bit careless for Apple to release Ventura without a finalized version of Xcode containing the corresponding SDK.

**Reply by smenor** [reply]
> I'm logged in with my Apple ID, but I'm unable to load the direct link to the Command Line Tools installer. It just redirects to my account page and says my Apple developer account is expired. Do you need a paid developer account to download this package?
> 
> The only other option I see is to download the full XCode @7+ GB
> 
> I've never had any problems with Homebrew until this incident. Now I need to access login protected dependency files just to run Homebrew? Seems like something went wrong in the planning process for macOS Venture 13 release.

I've used the macOS betas for years and always have the latest Xcode builds and still haven't been able to get this to work without trying to resort to weird mechanicians

**Reply by Bo98** [reply]
> It just redirects to my account page and says my Apple developer account is expired.

Must be a bug with expired accounts. Will probably work with a new account. But try this:

```
ADCAuthCookie=$(curl --cookie-jar - "https://developerservices2.apple.com/services/download?path=/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg")
curl --cookie <(echo "$ADCAuthCookie") -O "https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg"
```

**Reply by nwhitmont** [reply]
> Yeah, but not at Homebrew. Apple releasing an incomplete OS is something none of the Homebrew volunteers could have predicted

I feel like I just walked into one of the Apple spaceship glass doors.  Bad Apple!

Thank you Homebrew maintainers, y'all rock!

**Reply by mnapoli** [reply]
For some reason `brew upgrade` still asks to install CLI tools 😕  I'm still stuck.

**Reply by carlocab** [reply]
@mnapoli what is the output of `brew config` and `brew doctor`?

**Reply by mnapoli** [reply]
@carlocab both trigger the install of the CLI devtools. Here's the output: https://gist.github.com/mnapoli/4ddd1c839fd99fe06759b756d6eb4516

Edit: after `brew upgrade` I still get this error:

Error: Your Xcode (14.0.1) is too outdated.
Please update to Xcode 14.1 (or delete it).
Xcode can be updated from the App Store.

That is weird, I did exactly what was listed in the comment (installed 14.1 RC2).

**Reply by carlocab** [reply]
That's because you still have Xcode 14.0.1 installed. Your CLT installation is up-to-date, though.

Try doing
```sh
sudo xcode-select --switch /Library/Developer/CommandLineTools
```
If that doesn't work, try
```sh
sudo rm -rf /Applications/Xcode.app
```

**Reply by smenor** [reply]
> That's because you still have Xcode 14.0.1 installed. Your CLT installation is up-to-date, though.
> 
> 
> 
> Try doing
> 
> ```sh
> 
> sudo xcode-select --switch /Library/Developer/CommandLineTools
> 
> ```
> 
> If that doesn't work, try
> 
> ```sh
> 
> sudo rm -rf /Applications/Xcode.app
> 
> ```

I've done that and only have RC2 installed and it still fails for me and has failed for the last handful of Xcode βs but was working a couple of months back with the β Xcode and β macOS

**Reply by andrewcrook** [reply]
Download commandline tools for Xcode 14.1 RC 2 from Apple developer site.
open the dmg
and run the package 
after installing check
xcode-select -p           
is returning  /Library/Developer/CommandLineTools

Worked for me 

I have not bothered installing Xcode, until the official is out, command line tools package is independent.

**Reply by carlocab** [reply]
@smenor, sorry, my comment was addressed to @mnapoli, so it's not really a surprise it doesn't work for you. If you provide the output of `brew config` and `brew doctor`, as @mnapoli did, then that might help us work out what the problem is.

**Reply by smenor** [reply]
I think it's the same issue @carlocab

```sh
mbp-de-scott:~/ > brew config
HOMEBREW_VERSION: 3.6.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6a7eac25e167a1eb2d49e13c8cc530a3188af995
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4e2a09c6772dc07cdc7846908cacd4a3578158eb
Core tap last commit: 21 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code --wait
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.0.0.0.1.1661618636
Xcode: 14.1
Rosetta 2: false
```

```sh
mbp-de-scott:~/ > brew doctor 
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.1.


Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/smenor/.pyenv/shims/python2-config
  /Users/smenor/.pyenv/shims/python2.7-config
  /Users/smenor/.pyenv/shims/python-config

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  yosys
  six

Warning: Your Command Line Tools (CLT) does not support macOS 13.
It is either outdated or was modified.
Please update your Command Line Tools (CLT) or delete it if no updates are available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.1.
```

`brew config` literally says that I hav Xcode: 14.1 and that ∧ tells me to download it

**Reply by carlocab** [reply]
It's actually the opposite problem. Your `Xcode.app` is up-to-date, but your CLT is not:
```
CLT: 14.0.0.0.1.1661618636
```
Do
```
sudo rm -rf /Library/Developer/CommandLineTools
```
and download the latest "Command Line Tools for Xcode 14.1" from https://developer.apple.com/downloads/all. At the moment, that is "Command Line Tools for Xcode 14.1 Release Candidate 2".

**Reply by smenor** [reply]
Probably worth noting also that I'm almost always on the β stream for macOS and Xcode and this was working for awhile between a month or two after their post-WWDC release and a month or so ago

**Reply by smenor** [reply]
OK / done and finally working now.

Sorry @carlocab though for the record I've done this ∨ and followed the other instructions shown by brew many times to no avail so perhaps those should be updated ?

```sh
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install
```

**Reply by KenniJH** [reply]
> OK / done and finally working now.
> 
> 
> 
> Sorry @carlocab though for the record I've done this ∨ and followed the other instructions shown by brew many times to no avail so perhaps those should be updated ?
> 
> 
> 
> ```sh
> 
>   sudo rm -rf /Library/Developer/CommandLineTools
> 
>   sudo xcode-select --install
> 
> ```

Did exact same thing, works here as well

**Reply by andrewcrook** [reply]
@KenniJH 

does  `sudo xcode-select --install ` download command line tools for Xcode 14.1 RC2 because on Tuesday it certainly wasn’t. It had to be downloaded manually.

You don't actually need Xcode 14.1  just the command line tool they can work independently.

**Reply by jonathan-davies-linedup** [reply]
I'm unable to see / download the "Command Line Tools for Xcode 14.1" from https://developer.apple.com/downloads/all
Is this because I am not enrolled with Apple Developer?

Can anyone please help and share a download link, as I'm getting pretty desperate.

**Reply by p-linnane** [reply]
@jonathan-davies-linedup You can use the following provided by @Bo98:

```
ADCAuthCookie=$(curl --cookie-jar - "https://developerservices2.apple.com/services/download?path=/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg")
curl --cookie <(echo "$ADCAuthCookie") -O "https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg"
```

**Reply by jonathan-davies-linedup** [reply]
Thank you very much!

**Reply by KenniJH** [reply]
> @KenniJH
> 
> does `sudo xcode-select --install ` download command line tools for Xcode 14.1 RC2 because on Tuesday it certainly wasn’t. It had to be downloaded manually.
> 
> You don't actually need Xcode 14.1 just the command line tool they can work independently.

@andrewcrook

I'm not an expert, but i can state my situation and what eventually worked.

I had a working Hombrew install on MacOS Monterey and then i upgraded to MacOS Ventura.

Google led me to this discussion, since i had the same errors when trying to maintain my Homebrew install,
   - i.e. brew upgrade threw an error identical to the one mentioned OT.

I then removed the commandline tools folder/binaries ('rm -rf /Library/Developer/CommandLineTools') and installed it again ('xcode-select --install')

Did a 'brew doctor' ... fine
Did a 'brew -v'
 - that returned:
      Homebrew 3.6.7
      Homebrew/homebrew-core (git revision 755e2a7594e; last commit 2022-10-27)
      Homebrew/homebrew-cask (git revision 0af16f839b; last commit 2022-10-27)

Did a 'brew update && brew upgrade && brew upgrade --cask && brew autoremove && brew cleanup' 
 - That now worked, therefore my reply earlier :-)

You mentioned which version of Xcode commandline tools i ended up with and i just checked:
 - 'pkgutil --pkg-info=com.apple.pkg.CLTools_Executables' states:
      package-id: com.apple.pkg.CLTools_Executables
      version: **14.0.0.0.1.1661618636**
      volume: /
      location: /
      install-time: 1666876494
      groups: com.apple.FindSystemFiles.pkg-group

Then i installed some tools as a test:
 - 'brew install git'
 -  'brew install --cask visual-studio-code'

Worked out fine, so that's why I claim that I am up and running... Am I missing something?

**Reply by andrewcrook** [reply]
> Worked out fine, so that's why I claim that I am up and running... Am I missing something?
>

@KenniJH 

No not at all just working out what's going on and how things are developing since MacOS 13s release. My thinking was if the command tools are officially out via xcode-select --install then so is the full Xcode IDE and tools on App Store soon which I also use. Thanks

**Reply by nwhitmont** [reply]
Direct Link: https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg

**Reply by jtarchie** [reply]
You cannot access this link unless you are in the Apple Developer program. ^^^

**Reply by andrewcrook** [reply]
> You cannot access this link unless you are in the Apple Developer program. 

@jtarchie  I download it from the website. I have a developer account (using Apple ID) anyone can sign up free, however, I am not signed up to the Apple Developer program which you pay yearly.

**Reply by nwhitmont** [reply]
> You cannot access this link unless you are in the Apple Developer program. ^^^

I"m not a member of the developer program and I just copied the actual link from the Apple site after logging in with my regular AppleID.  

You do need to be logged in to developer.apple.com though at the time you use the link.  I'm guessing you weren't logged in to developer.apple.com when you tried it.

The real key to solving this scenario is running `brew config` to verify your Command Line Tools version. For example I installed the full XCode 14.1 but it didn't get my Command Line Tools to the desired version, which was confusing.  Running `brew config` showed me I still had the CLT 14.0.0.0.xxxx version, and installing the [Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2/Command_Line_Tools_for_Xcode_14.1_Release_Candidate_2.dmg) is what really solved the problem.

When the `brew doctor` tells you:
```
A newer Command Line Tools release is available
...
You should download the Command Line Tools for Xcode 14.1
```

This is actually slightly misleading as deleting CLT files and running `xcode-select --install` won't solve your problem, since the actual version you need is the Release Candidate (RC2) which won't be installed by the install command or the software update CLI command.

**Comment by andrewcrook** [comment]
Yeah you have to use the RC 2.  Xcode 14.1 hasn’t been officially release and , therefore , not on App Store either.
Ideally brew should have stuck with the current version of Xcode  until 14.1 is on the App Store and command line tools on xcode-select --install

**Reply by Bo98** [reply]
> Ideally brew should have stuck with the current version of Xcode until 14.1 is on the App Store and command line tools on xcode-select --install

We have for earlier macOS versions. For macOS 13, Xcode versions earlier than 14.1 do not contain the macOS 13 SDK.

**Reply by andrewcrook** [reply]
I am sure I can mostly run apps complied using the macOS12 SDK without having to recompile to macOS 13 SDK?
A lot of apps haven't been updated.
I think it is feature dependent. 
But yes this is Apple fault for not having consistent release dates.

**Reply by carlocab** [reply]
Note that software that Apple ships on Ventura (at least Safari, and everything in `/bin`) are built with the Ventura SDK, so it's definitely a good idea to follow Apple's lead here, even if it requires use of an RC.

**Comment by jasonmp85** [comment]
Welp, guess this is the thing to finally get me off the MAS version. Truly bizarre move by Apple (why not hold the release, or even just slap `14.1` on the RC and call it good?)

Maybe I missed something in the terminal output of brew, but I feel like having to figure this out may be duplicative for people not specifically developing apps for Apple hardware already…

**Comment by jaroslavas** [comment]
After reading all the posts I still get the same problem. Is there anything I can do?

`xcodebuild -version`:    
```                              
Xcode 14.1
Build version 14B47b
```

But `brew doctor` still gives me:
```
...
A newer Command Line Tools release is available
...
You should download the Command Line Tools for Xcode 14.1
```

And I still can't install some software: `brew install twittemb/formulae/Xcodecoverageconverter`

```
...
Error: Your Command Line Tools (CLT) does not support macOS 13.
...
```

**Reply by carlocab** [reply]
@jaroslavas: What is the output of `brew config`?

**Reply by jaroslavas** [reply]
```
brew config
HOMEBREW_VERSION: 3.6.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6a7eac25e167a1eb2d49e13c8cc530a3188af995
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e22841dbb48f3edc69740dc0d594db6552e2d95e
Core tap last commit: 4 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0-x86_64
CLT: 14.0.0.0.1.1661618636
Xcode: 14.1
```

**Reply by carlocab** [reply]
Same issue as @smenor, do https://github.com/Homebrew/discussions/discussions/3822#discussioncomment-3983390

**Comment by jasonplumb** [comment]
A Solution that worked for me:

> Initial warning: Warning: Your Xcode (14.0.1) is too outdated. Please update to Xcode 14.1 (or delete it). Xcode can be updated from the App Store.

1. brew install robotsandpencils/made/xcodes (here is the github page for more info: https://github.com/RobotsAndPencils/xcodes)
1. xcodes install '14.1 Release Candidate 2 (14B47b)'
1. brew doctor

> No warnings

**Reply by bjeanes** [reply]
Dang I was hopeful this would work for me, but I just get:

```sh-session
❯ xcodes install '14.1 Release Candidate 2 (14B47b)'
The network connection was lost.
```

Any subcommand gives that error even though I have network connectivity /shrug

**Reply by romanr** [reply]
Didn't know this exists, thank you! Perhaps brew should integrate this as a core command. Especially after this issue that can only be fixed by installing pre-release Xcode.

Nothing else worked for me, only this method:

```sh
brew install robotsandpencils/made/xcodes
brew install aria2 # optional for better downloading
xcodes install "14.1 Release Candidate 2 (14B47b)"
sudo xcode-select -s /Applications/Xcode-14.1.0-Release.Candidate.2.app/Contents/Developer
```

**Reply by jasonplumb** [reply]
@bjeanes Try `xcodes list` and then reference another version of `14.1`

**Reply by jasonplumb** [reply]
@romanr yeah I agree. I have always felt that the need to install xcode when updating brew was a bit strange so the ability to switch to different versions it a game changer

**Reply by andrewcrook** [reply]
Brew only requires the command line tools for Xcode witch are interdependent from the main Xcode package.
As far as I know xcodes doesn’t switch versions of the command line tools only the visual environment. 

> `sudo xcode-select -s /Applications/Xcode-14.1.0-Release.Candidate.2.app/Contents/Developer`

If you have installed command line tools correctly it should be 

```
sudo xcode-select  --install 
^^^ normally install like this, however, at moment you have to download the RC from developer.apple.com for free
sudo xcode-select -s /Library/Developer/CommandLineTools
```

**Reply by andrewcrook** [reply]
and once installed command line tools should keep its self updated automatically via macOS’s software update with OS and security updates (not the App Store)

**Reply by alsolonets** [reply]
In my case I was also required to run:
```
sudo rm -rf /Library/Developer/CommandLineTools
```
prior to running 
```
sudo xcode-select -s /Library/Developer/CommandLineTools
```

**Comment by vikjam** [comment]
What are the consequences of `Warning: Your Command Line Tools (CLT) does not support macOS 13.`? It only seems to appear for me with `brew doctor`, but I seem to be able to `brew install` all that I need. What problems can arise with an outdated CLT install?

**Reply by SMillerDev** [reply]
If you install something (like python) that depends on the CLT and you have a different version it might not work.

**Comment by Bo98** [comment]
Xcode 14.1 is now released. It is identical to RC2 so if you installed that, you already have the final version.

**Comment by HermannGruber** [comment]
For those who used the following temporary workaround: https://github.com/orgs/Homebrew/discussions/3822#discussioncomment-3965318
After you've updated the official version of Xcode via the App Store, don't forget to switch back to the official version of Xcode: 
`sudo xcode-select --switch /Applications/Xcode.app`
This will bring your Xcode app back onto track for future updates - security updates among other.

**Reply by iambudi** [reply]
Brew doctor says its Xcode that outdated not Command Line Tool.
```bash
Warning: Your Xcode (14.0.1) is too outdated.
Please update to Xcode 14.1 (or delete it).
Xcode can be updated from the App Store.
```
 
Looks like i have to download Xcode 14.1 with size is 7.05 GB just for the brew to be able to do package installation normally.

**Reply by Bo98** [reply]
You can delete the old Xcode if you don't need/want it. `brew` will be able to use the CLT only for most formulae (except Qt etc.)

> brew to be able to do package installation normally

Installing bottles works. It's just building from source which, unsurprisingly, needs development tools and thus requires ones compatible with your version of macOS.

**Reply by iambudi** [reply]
I already replaced the old one with Xcode 14.1. Now the warning is gone.

**Reply by andrewcrook** [reply]
@iambudi 

> Brew doctor says its Xcode that outdated not Command Line Tool.
> Looks like i have to download Xcode 14.1 with size is 7.05 GB just for the brew to be able to do package installation normally.

You don't _have_ to install `Xcode 14.01` you can install the` command line tools for macOS 14.1` both work inderpendenty and contain all the command line components which brew uses.

if fact I was using just the RC command line tools until the full Xcode 14.1 was released officially on the App Store.

**command line tools** - required tools for brew
```
xcode-select --install
sudo xcode-select -s /Library/Developer/CommandLineTools
```
Note,  the above is normally automatically updated via software update 

**Complete Xcode** - Everything, the IDE, simulators, extensions etc etc
- install from App Store 
```
sudo xcode-select -s /Applications/Xcode.app
```
Automatically updated by App Store

**Comment by pboese** [comment]
I'm having a similar issue when running `brew upgrade` or `brew doctor`

```
Error: Your Xcode (14.2 => /Volumes/Apps/Applications/Xcode.app/Contents/Developer) is too outdated.
Please update to Xcode 15.0 (or delete it).
Xcode can be updated from the App Store.
```

But Command Line Tools 15 are installed:

```
package-id: com.apple.pkg.CLTools_Executables
version: 15.0.0.0.1.1694021235
volume: /
location: /
install-time: 1702385733
```

**Reply by andrewcrook** [reply]
IIRC Homebrew now requires Xcode as well as Xcode command line tools

**Reply by Bo98** [reply]
Homebrew can operate without Xcode.app for most cases, but if it's present it must not be too old as it's difficult to stop things using it.

In your case you do have a too old Xcode.app that will not launch on your macOS.

**Reply by andrewcrook** [reply]
@Bo98  Thanks for verifying its been some time since I last had that error
